classification_analysis.ipynb

```diff
@@ -224,45 +264,65 @@
     "        for phone_label, phone_detail_map in phone_map.items():\n",
     "            if \"control\" in phone_detail_map[\"role\"]:\n",
     "                continue\n",
-    "            for r in phone_detail_map[\"evaluation_ranges\"]:\n",
+    "            for run_ix, r in enumerate(phone_detail_map[\"evaluation_ranges\"]):\n",
     "                if r['eval_role_base'] != role:\n",
     "                    continue\n",
+    "\n",
     "                tr_ss  = []\n",
     "                tr_gts = []\n",
+    "                gt_traj = []\n",
+    "                \n",
     "                for i, tr in enumerate(r[\"evaluation_trip_ranges\"]):\n",
     "                    for ss in tr[\"sensed_section_ranges\"]:\n",
-    "                        ## append the sensed section data\n",
     "                        tr_ss.append(ss)\n",
     "                    for section in tr[\"evaluation_section_ranges\"]:\n",
+    "\n",
     "                        ## get the ground truth section data\n",
     "                        section_gt_leg = pv.spec_details.get_ground_truth_for_leg(tr['trip_id_base'],\n",
     "                                                                                  section['trip_id_base'],\n",
     "                                                                                  tr['start_ts'],\n",
     "                                                                                  tr['end_ts'])\n",
     "                        \n",
-    "                        if section_gt_leg[\"type\"] == \"WAITING\": \n",
+    "                        if section_gt_leg[\"type\"] == \"WAITING\":\n",
     "                            continue\n",
+    "                        \n",
     "                        gts = {'start_ts': section['start_ts'], \n",
     "                               'end_ts': section['end_ts'], \n",
-    "                               'mode': section_gt_leg['mode']}\n",
+    "                               'mode': section_gt_leg['mode']\n",
+    "                               }\n",
+    "\n",
     "                        tr_gts.append(gts)\n",
+    "\n",
+    "                    trajectory_data = get_reference_trajectory(pv.spec_details.CURR_SPEC_ID, tr['trip_id_base'])\n",
+    "                    run_id = run_ix\n",
+    "                    trajectory_data = trajectory_data.loc[\n",
+    "                        (trajectory_data.source == os) &\n",
+    "                        (trajectory_data.run == run_id), :\n",
+    "                        ].reset_index(drop=True, inplace=False)\n",
```

Comment:
We reset indices so that subsequent downstream list slicing operations do not cause indexing issues. In my experience, not resetting indices after subsetting data frames is the culprit in data processing pipelines.

However, that being said, we don't subset the data frame anymore after this, so not issuing .reset_index would be fine.

Reply:
I am fine with resetting, just to be on the safe side. But you might want to add a comment to this effect.